python/fbprophet/diagnostics.py

```diff
@@ -105,43 +110,84 @@ def cross_validation(model, horizon, period=None, initial=None):
     if model.uncertainty_samples:
         predict_columns.extend(['yhat_lower', 'yhat_upper'])
 
+    predicts=[]
+
     cutoffs = generate_cutoffs(df, horizon, initial, period)
-    predicts = []
-    for cutoff in cutoffs:
-        # Generate new object with copying fitting options
-        m = prophet_copy(model, cutoff)
-        # Train model
-        history_c = df[df['ds'] <= cutoff]
-        if history_c.shape[0] < 2:
-            raise Exception(
-                'Less than two datapoints before cutoff. '
-                'Increase initial window.'
-            )
-        m.fit(history_c, **model.fit_kwargs)
-        # Calculate yhat
-        index_predicted = (df['ds'] > cutoff) & (df['ds'] <= cutoff + horizon)
-        # Get the columns for the future dataframe
-        columns = ['ds']
-        if m.growth == 'logistic':
-            columns.append('cap')
-            if m.logistic_floor:
-                columns.append('floor')
-        columns.extend(m.extra_regressors.keys())
-        columns.extend([
-            props['condition_name']
-            for props in m.seasonalities.values()
-            if props['condition_name'] is not None])
-        yhat = m.predict(df[index_predicted][columns])
-        # Merge yhat(predicts), y(df, original data) and cutoff
-        predicts.append(pd.concat([
-            yhat[predict_columns],
-            df[index_predicted][['y']].reset_index(drop=True),
-            pd.DataFrame({'cutoff': [cutoff] * len(yhat)})
-        ], axis=1))
+
+    if multiprocess is True:
+        with Pool() as pool:
+            logger.info('Running cross validation in multiprocessing mode')
+            input_df = [[df, model, cutoff, horizon, predict_columns] for cutoff in cutoffs]
+            predicts = pool.starmap(single_cutoff_forecast, input_df)
+    elif multiprocess is None:
+        predicts = []
+        for cutoff in cutoffs:
+            predicts.append(single_cutoff_forecast(df, model, cutoff, horizon, predict_columns))
+    else:
+        raise ValueError(f"{multiprocess} is not a valid assignment to multiprocess argument.Valid "
+                         f"options are 'None' or 'True'")
```

Comment:
I think this wouldn't be necessary if it was True/False options for multiprocessing, it'd be more apparent